src/components/windows-motion-controls.js

```diff
@@ -0,0 +1,382 @@
+var bind = require('../utils/bind');
+var registerComponent = require('../core/component').registerComponent;
+var controllerUtils = require('../utils/tracked-controls');
+var utils = require('../utils/');
+
+var debug = utils.debug('components:windows-motion-controls:debug');
+var warn = utils.debug('components:windows-motion-controls:warn');
+
+// TODO: Point to final glb assets once they are hosted. For now, place them in this hard coded directory.
+var MODEL_BASE_URL = '/examples/assets/models/controllers/wmr/';
+var MODEL_LEFT_FILENAME = 'left.glb';
+var MODEL_RIGHT_FILENAME = 'right.glb';
+var MODEL_UNIVERSAL_FILENAME = 'universal.glb';
+
+var GAMEPAD_ID_PREFIX = 'Spatial Controller (Spatial Interaction Source) ';
+var GAMEPAD_ID_PATTERN = /([0-9a-zA-Z]+-[0-9a-zA-Z]+)$/;
+
+/**
+ * Windows Motion Controller Controls Component
+ * Interfaces with Windows Motion Controller controllers and maps Gamepad events to
+ * common controller buttons: trackpad, trigger, grip, menu and system
+ * It loads a controller model and transforms the pressed buttons
+ */
+module.exports.Component = registerComponent('windows-motion-controls', {
+  schema: {
+    hand: {default: 'left'},
+    // If true, loads the controller glTF asset.
+    model: {default: true},
+    // If true, will hide the model from the scene if no matching gamepad (based on ID & hand) is connected.
+    hideDisconnected: {default: true}
+  },
+
+  _mapping: {
```

Comment:
We don't use underscore to indicate private variables. Can we keep just `mapping`?